actor/actor.go

```diff
@@ -78,25 +81,40 @@ func (b *ServerImplBase) SetID(id string) {
 
 // SaveState is to saves the state cache of this actor instance to state store component by calling api of daprd.
 func (b *ServerImplBase) SaveState() error {
+	return b.SaveStateContext(context.Background())
+}
+
+func (b *ServerImplBase) SaveStateContext(ctx context.Context) error {
 	if b.stateManager != nil {
-		return b.stateManager.Save()
+		return b.stateManager.SaveContext(ctx)
 	}
 	return nil
 }
 
+//nolint:interfacebloat
 type StateManager interface {
 	// Add is to add new state store with @stateName and @value
 	Add(stateName string, value interface{}) error
+	// Add is the same as Add, but with context.
+	AddContext(ctx context.Context, stateName string, value interface{}) error
```

Comment:
I would like `AddWithContext()` to include `Add()`, or make `AddContext()`  only to do one thing -- add context.

Reply:
Would it make more sense to deprecate this interface and create a new one? The existing one would still be supported but fallback to calling the new implementation where context is always present.

Reply:
I agree.

@JoshVanL please update as @artursouza said.